### PR TITLE
RoboCup 2019 make vision multicast address a parameter

### PR DESCRIPTION
**Pull request by MathewMacDougall**
### Description
Changes from RoboCup 2019. Updated launch files to restart nodes, and made the vision multicast address optionally set by an environment variable.

### Testing Done
unit tests still pass, validated with RoboCup gameplay

### Resolved Issues
None

### Length Justification
None

### CODE REVIEW

File: src/thunderbots/software/launch/ai.launch [hunk 1]
```diff
@@ -4,19 +4,24 @@
         Runs all of the AI nodes except for the backend (Grsim, MRF, etc.)
     -->
 
+    <param name="vision_multicast_address" value="$(optenv VISION_MULTICAST_ADDRESS 224.5.23.2)"/>
+
+    <!-- Start rosbag -->
+    <node name="rosbag_record_all" pkg="rosbag" type="record" args="-a -o $(find thunderbots)"/>
```
Comment (henryymliu) [inline src/thunderbots/software/launch/ai.launch]: Do we still want to keep rosbag recording by default?

Comment (garethellis0) [reply src/thunderbots/software/launch/ai.launch]: I don't think it makes a huge difference, but I'm inclined to keep whatever we do in master right now (so don't record).

File: src/thunderbots/software/launch/ai.launch [hunk 2]
```diff
@@ -4,19 +4,24 @@
         Runs all of the AI nodes except for the backend (Grsim, MRF, etc.)
     -->
 
+    <param name="vision_multicast_address" value="$(optenv VISION_MULTICAST_ADDRESS 224.5.23.2)"/>
```
Comment (garethellis0) [inline src/thunderbots/software/launch/ai.launch:7]: When I try running `optenv VISION_MULTICAST_ADDRESS 224.5.23.2` locally I get `optenv: command not found`?

File: src/thunderbots/software/network_input/networking/network_client.cpp [hunk 3]
```diff
@@ -23,7 +23,7 @@ NetworkClient::NetworkClient(ros::NodeHandle& node_handle)
     // Set up our connection over udp to receive vision packets
     try
     {
-        std::string vision_multicast_address;
+        std::string vision_multicast_address = "224.5.23.2";
```
Comment (garethellis0) [inline src/thunderbots/software/network_input/networking/network_client.cpp]: Is this address somehow meaningful? Is it the EDC address, or maybe the one at competition? We should give it a name, like `RANDOM_MULTICAST_ADDRESS` or `UBC_EDC_BAY_MULTICAST_ADDRESS`

**Review by garethellis0 — CHANGES_REQUESTED** [review]

File: src/thunderbots/software/launch/ai.launch (hunk 2 above)
Comment (henryymliu) [reply src/thunderbots/software/launch/ai.launch:7]: When you say locally do you mean running it in terminal? Looks like optenv is a ROS specific thing. http://wiki.ros.org/roslaunch/XML

File: src/thunderbots/software/network_input/networking/network_client.cpp (hunk 3 above)
Comment (henryymliu) [reply src/thunderbots/software/network_input/networking/network_client.cpp]: It's the default address used by SSL Vision, and by extension at competition and on our field. Something like DEFAULT_MULTICAST_ADDRESS would be fine with me

Comment (garethellis0) [reply src/thunderbots/software/network_input/networking/network_client.cpp]: Ah, maybe `ROBOCUP_SSL_DEFAULT_MULTICAST_ADDRESS`?

File: src/thunderbots/software/launch/ai.launch (hunk 2 above)
Comment (garethellis0) [reply src/thunderbots/software/launch/ai.launch:7]: Wow yup ok nvm. All good.

File: src/thunderbots/software/launch/ai.launch (hunk 1 above)
Comment (MathewMacDougall) [reply src/thunderbots/software/launch/ai.launch]: Removed rosbag from launchfile.

File: src/thunderbots/software/network_input/networking/network_client.cpp (hunk 3 above)
Comment (MathewMacDougall) [reply src/thunderbots/software/network_input/networking/network_client.cpp]: Changed to `SSL_VISION_DEFAULT_MULTICAST_ADDRESS`

**Review by garethellis0 — APPROVED** [review]
LGTM 👍 🎉

**Pull request merged**